src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -91,13 +95,93 @@ impl ScalarFunc {
         };
         Ok(Some(Cow::Owned(sha2)))
     }
+
+    #[inline]
+    pub fn compress<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        if input.is_empty() {
+            return Ok(Some(Cow::Borrowed(b"")));
+        }
+        let mut e = ZlibEncoder::new(input.as_ref(), Compression::default());
+        let mut vec = Vec::with_capacity(1024);
+        match e.read_to_end(&mut vec) {
+            Ok(len) => {
+                let mut cap = 4 + len;
+                // according to MySQL doc: If the string ends with space,
+                // an extra . character is added to avoid problems with endspace trimming
+                if vec[len - 1] == 32 {
+                    vec.push(b'.');
+                    cap += 1;
+                }
+
+                let mut wtr = Vec::with_capacity(cap);
+                wtr.write_u32::<LittleEndian>(input.len() as u32).unwrap();
+                wtr.extend_from_slice(&vec);
```

Comment:
Personally I prefer to write `use byteorder::xxx` at the scope of this function, because it is not useful for the rest of the file. Also apply to `uncompress`.

Reply:
updated. PTAL
but now there are `use byteorder::{ByteOrder, LittleEndian};` in three functions of the single file :(